pocs/dome/astrohaven.py

```diff
@@ -0,0 +1,127 @@
+# Based loosely on the code written by folks at Wheaton College, including:
+# https://github.com/goodmanj/domecontrol
+
+import time
+
+from pocs.dome import abstract_serial_dome
+
+
+class Protocol:
+    # Response codes
+    BOTH_CLOSED = '0'
+    BOTH_OPEN = '3'
+
+    # TODO(jamessynge): Confirm and clarify meaning of '1' and '2'
+    B_IS_OPEN = '1'
+    A_IS_OPEN = '2'
+
+    A_OPEN_LIMIT = 'x'  # Response to asking for A to open, and being at open limit
+    A_CLOSE_LIMIT = 'X'  # Response to asking for A to close, and being at close limit
+
+    B_OPEN_LIMIT = 'y'  # Response to asking for B to open, and being at open limit
+    B_CLOSE_LIMIT = 'Y'  # Response to asking for B to close, and being at close limit
+
+    # Action codes, echoed while happening
+    CLOSE_A = 'A'
+    OPEN_A = 'a'
+
+    CLOSE_B = 'B'
+    OPEN_B = 'b'
+
+    OPEN_BOTH = 'O'
+    CLOSE_BOTH = 'C'
+    RESET = 'R'
+
+
+class AstrohavenDome(abstract_serial_dome.AbstractSerialDome):
+    """Interface to an Astrohaven clamshell dome with a Vision 130 PLC and RS-232 interface.
+
+    Experience shows that it emits a status byte about once a second, with the codes
+    documented about in the Protocol class.
+    """
+    LISTEN_TIMEOUT = 3  # Max number of seconds to wait for a response
+    MOVE_TIMEOUT = 10  # Max number of seconds to run the door motors
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # TODO(jamessynge): Consider whether to expose settings of the pyserial object thru
+        # rs232.SerialData. Probably should. Could use newer dictionary get/set mechanism so
+        # that change to SerialData is minimal. Alternately, provide a means of reading
+        # that info from the config file in AbstractSerialDome.__init__ and using it to
+        # initialize the SerialData instance.
+
+        # Let's use a timeout that is long enough so that we are "guaranteed" a byte of output
+        # from the device. 1 second seems too small given that it appears that is the pace of
+        # output from the PLC.
+        self.ser.ser.timeout = AstrohavenDome.LISTEN_TIMEOUT
+
+    @property
+    def is_open(self):
+        v = self.read_latest_state()
+        return v == Protocol.BOTH_OPEN
+
+    def open(self):
+        self.full_move(Protocol.OPEN_A, Protocol.A_OPEN_LIMIT)
+        self.full_move(Protocol.OPEN_B, Protocol.B_OPEN_LIMIT)
+        return self.is_open
+
+    @property
+    def is_closed(self):
+        v = self.read_latest_state()
+        return v == Protocol.BOTH_CLOSED
+
+    def close(self):
+        self.full_move(Protocol.CLOSE_A, Protocol.A_CLOSE_LIMIT)
+        self.full_move(Protocol.CLOSE_B, Protocol.B_CLOSE_LIMIT)
+        return self.is_closed
+
+    @property
+    def state(self):
```

Comment:
Comment in other PR was about switching this to `status`. You seem to be using `read_latest_state` for all decision making so this seems like it is for a user.

Reply:
Yes, it is for the end user. Will create another issue/branch/PR for general changes to the dome API.

Reply:
Filed issue #189 for this.